fix(GiniBankSDK): Expose cancel sdk function

igor-gini opened the pull request:
PP-129

ValentinaIancu-Gini commented on BankSDK/GiniBankSDKExample/GiniBankSDKExample/DemoViewController.xib:
```diff
@@ -117,26 +118,37 @@
                     <nil key="textColor"/>
                     <nil key="highlightedColor"/>
                 </label>
+                <stackView opaque="NO" contentMode="scaleToFill" spacing="16" translatesAutoresizingMaskIntoConstraints="NO" id="QRX-Mq-lHn">
```
[clarification]Why did we not add this new option in the Settings screen? We will have more things to fix when we support landscape mode and as well in the Photo payment team we will add a new button to see the "transaction list". I would suggest moving it inside the Settings screen at the button of the screen where all the "DEbug and development options" are located.

igor-gini replied:
Added a Settings item

ValentinaIancu-Gini reviewed (commented):
@igor-gini I finished the code review for this pr. Can you please check my suggestions and questions? Thank you!

zladzeyka reviewed (changes requested):
@igor-gini please add also a unit test

igor-gini commented on BankSDK/GiniBankSDK/Sources/GiniBankSDK/Core/GiniBankNetworkingScreenApiCoordinator.swift:
```diff
@@ -15,6 +15,7 @@ protocol Coordinator: AnyObject {
 }
 
 open class GiniBankNetworkingScreenApiCoordinator: GiniScreenAPICoordinator, GiniCaptureDelegate {
+    static var currentCoordinator: GiniBankNetworkingScreenApiCoordinator?
```
This needs to be at least at internal level so tests would be able to access it.

igor-gini commented on BankSDK/GiniBankSDK/Sources/GiniBankSDK/Core/Networking/GiniBank+Networking.swift:
```diff
@@ -10,6 +10,12 @@ import UIKit
 import GiniBankAPILibrary
 extension GiniBank {
+    /**
+     Aborts the current GiniBankSDK session
```
Sure, added

igor-gini commented on BankSDK/GiniBankSDK/Sources/GiniBankSDK/Core/GiniBankNetworkingScreenApiCoordinator.swift:
```diff
@@ -284,6 +286,15 @@ open class GiniBankNetworkingScreenApiCoordinator: GiniScreenAPICoordinator, Gin
         return self.start(withDocuments: documents, animated: animated)
     }
+
+    public func closeSDK() {
```
I made it public, so a particular instance could be cancelled from anywhere, but I suppose it’s not actually needed, replaced it with calling finishWithCancellation directly, followed by cleaning up GiniBankConfiguration(ScreenAPICoordinator seems to be doing this in closeSreenAPIAndSendTransferSummary, so I guess we also need that here).

igor-gini commented:
> @igor-gini please add also a unit test

I added a test, but it passes only if GiniBankAPI does not trigger preconditionFailure that there is no entitlement for access to Keychain. I added a possible solution: I made the global internal variable _GINIBANKAPILIBRARY_DISABLE_KEYCHAIN_PRECONDITION_FAILURE and set it to true in the test.

zladzeyka commented on BankAPILibrary/GiniBankAPILibrary/Sources/GiniBankAPILibrary/Documents/Core/GiniBankAPI.swift:
```diff
@@ -7,6 +7,9 @@
 
 import Foundation
 
+// should only be used in tests
```
[Suggestion] Please add a comment with more explanation like:

```
 /**
     Sets the default error logger. It is only used when giniErrorLoggerIsOn is true.

     - note: Internal usage only.
   */
```

igor-gini replied:
Sure, done

zladzeyka reviewed (changes requested):
Thank you for the update @igor-gini 
2 tiny suggestions from my side :)
